### PR TITLE
File creator

### PR DESCRIPTION
**Pull request by s-taylor**
This PR splits out the scaffolding code that was repeated in base, controller and cors into a separate lib file. I feel like it could perhaps be more DRY still, so open to suggestions anyone has.

This should resolve https://github.com/Prismatik/redbeard/issues/13

### CODE REVIEW

File: base/index.js [hunk 1]
```diff
@@ -28,21 +28,9 @@ module.exports = (opts) => {
...
-
-  things.forEach(thing => {
-    const templatePath = path.join(__dirname, thing.p, thing.n+'.mustache');
-    const template = fs.readFileSync(templatePath).toString();
-
-    const dir = path.join(process.cwd(), thing.d);
-    const target = path.join(dir, thing.t+'.'+thing.e);
-
-    if (fs.existsSync(target)) return console.error(target, 'already exists. Not overwriting it');
+  ].map(file => _.extend({p: '', d: file.p || '', t: file.n, e: 'js'}, file));
```
Comment (nwinch) [inline base/index.js]: I think we should move this `map` and `extend` into the `scaffold` lib, as very similar in all three places.

File: lib/scaffold.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const fs = require('fs');
+const path = require('path');
+const mkdirp = require('mkdirp');
+const mustache = require('mustache');
+
+// takes a .mustache file and renders this to a target folder with a given
+// filename and file extension
+module.exports = (mustacheOpts, basePath) => file => {
```
Comment (nwinch) [inline lib/scaffold.js]: I think have `scaffold` receive an `opts` object which could have:
- `opts.files (Array of files)`
- `opts.basePath`
- `opts.mustacheOpts`

And then move any of the file mapping/extend logic into this function.

Also maybe use a named function for better error handling?

**Comment by nwinch** [comment]
Should we add tests for this functionality?

**Comment by nwinch** [comment]
@s-taylor Review done with some feedback.

**Comment by s-taylor** [comment]
@nwinch PR has now been updated as per your feedback.

I've also added redbeard tests to the project using Mocha and Must (rather than just tests for our generated code which is what we had prior to this). This means our test suite is running both blue-tape tests and also mocha tests which is a little gross, but we have a separate task pending to refactor all our redbeard generated tests to Mocha.

This will need another review as there are a lot of changes.

File: test/lib/scaffold_test.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    });
+
+    it('must create destination folder', function() {
+      scaffold({
+        basePath: sourcePath,
+        files: [{n: 'scaffold', d: outputFolder}],
+        mustacheOpts: {}
+      });
+      
+      const folder= fs.lstatSync(outputPath);
```
Comment (s-taylor) [inline test/lib/scaffold_test.js]: note to self, add space `const foldler =`

**Comment by nwinch** [comment]
LGTM 👍

**Pull request merged**